### PR TITLE
chore: bump @gmod dependencies and generic-filehandle

### PR DESCRIPTION
**Pull request by manzt**
Closes #807

## Change List
 - bump `@gmod/*` packages
 - bump `generic-filehandle`
 - add `optimizeDeps.esbuildOptions.inject` to inject `Buffer` polyfill globally for all dependencies. 

## Checklist
 - [ ] Ensure the PR works with all demos on the online editor
 - [ ] Unit tests added or updated
 - [ ] Examples added or updated
 - [ ] [Documentation](https://github.com/gosling-lang/gosling-website) updated (e.g., added API functions)
 - [ ] Screenshots for visual changes (e.g., new encoding support or UI change on Editor)

### CODE REVIEW

**Comment by manzt** [comment]
seem to be issue with bigwig files ..

**Comment by manzt** [comment]
Ok I updated the `vite.config.js` to inject the `Buffer` into the dependencies.. This isn't ideal. Really, upstream we should have `@gmod/*` have explicit imports for `Buffer` in the library code:

```
import { Buffer } from 'buffer';
```

instead of just relying on a global. This is easier for bundlers to resolve, and something that I implemented a while ago in `generic-filehandle`.

https://github.com/GMOD/generic-filehandle/pull/102

**Comment by manzt** [comment]
@sehilyi it seems adding those `import { Buffer } from "buffer"` is welcomed from the @GMOD folks. I won't be able to make this PR for a bit FYI

**Comment by sehilyi** [comment]
Thanks for letting me know. I might not be able to find time this week but probably possible next week.

File: src/data-fetchers/bigwig/bigwig-data-fetcher.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ function BigWigDataFetcher(HGC: import('@higlass/types').HGC, dataConfig: BigWig
             this.dataPromises.push(this.loadBBI(dataConfig));
         }
 
-        loadBBI(dataConfig: { url: string }) {
+        async loadBBI(dataConfig: { url: string }) {
```
Comment (sehilyi) [inline src/data-fetchers/bigwig/bigwig-data-fetcher.ts:89]: I assume this change was needed by some recent changes in BBI?

**Review by sehilyi — APPROVED** [review]
Thanks for this change -- tested locally with demo examples and works well

**Comment by sehilyi** [comment]
@manzt Now there is a conflict in `yarn.lock`. Should we regenerate the lock file?

**Comment by manzt** [comment]
yah we can just copy the lock file over from master or relock.

```
git checkout master -- yarn.lock
```

File: src/data-fetchers/bigwig/bigwig-data-fetcher.ts (hunk 1 above)
Comment (manzt) [reply src/data-fetchers/bigwig/bigwig-data-fetcher.ts:89]: Not exactly. This function returns either a promise or `null`, and we push the results to `dataPromises`. Previously the return type was `Promise<void> | null`. The async makes it always return a Promise.

```ts
        async loadBBI(dataConfig: { url: string }) {
            if (dataConfig.url) {
                this.bwFile = new BigWig({
                    filehandle: new RemoteFile(dataConfig.url)
                });
                return this.bwFile.getHeader().then((h: BigWigHeader) => {
                    this.bwFileHeader = h;
                });
            } else {
                console.error('Please enter a "url" field to the data config');
                return null;
            }
        }
```

**Comment by manzt** [comment]
ok, I rebased the changes @sehilyi

**Comment by sehilyi** [comment]
Thanks! I will go ahead and merge this.

File: src/data-fetchers/bigwig/bigwig-data-fetcher.ts (hunk 1 above)
Comment (sehilyi) [reply src/data-fetchers/bigwig/bigwig-data-fetcher.ts:89]: I see – Good to know that.

**Pull request merged**